### PR TITLE
Additions to the Locks API

### PR DESCRIPTION
**Pull request by landreev**
**What this PR does / why we need it**:
This is the code part of IQSS/dataverse.harvard.edu#135, the local issue opened for dealing with the backlog of dataset locks in production. 
This PR extends the existing locks API by adding an end point for listing all the locks that are present (the old API is limited to checking for locks on specific datasets). Going forward this API can be used to monitor locks without having to run database queries. 
Listing can be filtered by lock type and user with the optional query parameters. 

**Which issue(s) this PR closes**:

Closes IQSS/dataverse.harvard.edu#135

**Special notes for your reviewer**:
The new API is superuser-only. I briefly considered placing it under /api/admin, but decided to keep it under /api/datasets with the existing locks APIs. In practice we will want this new API to be used by admin users who may not have access to localhost. 
The rest of it is very straightforward.

**Suggestions on how to test this**:
Should be self-explanatory based on the documentation section in the API guide. 

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

File: doc/sphinx-guides/source/api/native-api.rst [hunk 1]
```diff
@@ -1612,20 +1617,43 @@ Or, to delete a lock of the type specified only. Note that this requires “supe
...
+The listing can be filtered by specific lock type **and/or** user, using the following *optional* query parameters:
+
+* ``userIdentifier`` - To list the locks owned by a specific user
+* ``type`` - To list the locks of the type specified. If the supplied value does not match a known lock type, the API will return an error and a list of valid lock types. As of writing this, the implemented lock types are ``Ingest``, ``Workflow``, ``InReview``, ``DcmUpload``, ``finalizePublication``, ``EditInProgress`` and ``FileValidationFailed``.
+
+For example:
+
+.. code-block:: bash
+
+  curl -H "X-Dataverse-key: xxx" "http://localhost:8080/api/datasets/locks?type=Ingest&userIdentifier=davis4ever"
```
Comment (sekmiller) [inline doc/sphinx-guides/source/api/native-api.rst:1654]: davis4ever

Comment (landreev) [reply doc/sphinx-guides/source/api/native-api.rst:1654]: Good eye! 😄

**Review by sekmiller — APPROVED** [review]
looks good. moving to QA

**Pull request merged**